libs/wingsdk/src/target-tf-azure/function.ts

```diff
@@ -0,0 +1,207 @@
+      name: `${functionName}.zip`,
+      storageAccountName: this.storageAccount.name,
+      storageContainerName: functionCodeBucket.storageContainer.name,
+      type: "Block",
+      source: asset.path,
+    });
+
+    // Create the function
+    this.function = new LinuxFunctionApp(this, "Function", {
+      name: functionName,
+      resourceGroupName: this.resourceGroup.name,
+      location: this.resourceGroup.location,
+      servicePlanId: this.servicePlan.id,
+      storageAccountName: this.storageAccount.name,
+      identity: {
+        type: functionIdentityType,
+      },
+      storageAccountAccessKey: this.storageAccount.primaryAccessKey,
+      siteConfig: {
+        applicationStack: {
+          nodeVersion: functionNodeVersion,
+        },
+      },
+      httpsOnly: true,
+      appSettings: {
+        ...this.env,
+        WEBSITE_RUN_FROM_PACKAGE: `https://${this.storageAccount.name}.blob.core.windows.net/${functionCodeBucket.storageContainer.name}/${functionCodeBlob.name}`,
+        FUNCTIONS_WORKER_RUNTIME: functionRuntime,
+      },
+    });
+
+    // Add permissions to read function code
+    new RoleAssignment(this, `ReadLambdaCodeAssignment`, {
+      principalId: this.function.identity.principalId,
+      roleDefinitionName: StorageAccountPermissions.READ,
+      scope: this.storageAccount.id,
+    });
+
+    // Apply permissions from bound resources
+    for (const permissionScope of this.permissions?.keys() || []) {
+      for (const roleDefinitionName of this.permissions?.get(permissionScope) ??
+        []) {
+        new RoleAssignment(this, `RoleAssignment${roleDefinitionName}`, {
+          scope: permissionScope,
+          roleDefinitionName,
+          principalId: this.function.identity.principalId,
+        });
+      }
+    }
+  }
+
+  /**
+   *  Adds role to function for given azure scope
+   *
+   * @param scope - The azure permission scope of the role assignment to create.
+   * @param roleDefinitionName - The name of the role definition to use in the role assignment.
+   */
+  public addPermission(scope: string, roleDefinitionName: string) {
+    if (!this.permissions) {
+      this.permissions = new Map();
+    }
+
+    // If the function has already been initialized attach the role assignment directly
+    if (this.function) {
+      if (
+        this.permissions.has(scope) &&
+        this.permissions.get(scope)?.has(roleDefinitionName)
+      ) {
+        return; // already exists
+      }
+
+      new RoleAssignment(this, `RoleAssignment${roleDefinitionName}`, {
+        scope,
+        roleDefinitionName,
+        principalId: this.function.identity.principalId,
+      });
+    }
+
+    const roleDefinitions = this.permissions.get(scope) ?? new Set();
+    roleDefinitions.add(roleDefinitionName);
+    this.permissions.set(scope, roleDefinitions);
+  }
+
+  /** @internal */
+  public _toInflight(): core.Code {
+    throw new Error("Method not implemented.");
```

Comment:
Maybe we can add a more specific error message like "cloud.Function can't be referenced inflight on Azure yet" and link the issue: https://github.com/winglang/wing/issues/1371